internal/cmd/controller/controllers/clusterregistration/controller.go

```diff
@@ -191,18 +191,18 @@ func (h *handler) OnChange(request *fleet.ClusterRegistration, status fleet.Clus
 
 	logrus.Infof("Cluster registration request '%s/%s', cluster '%s/%s' granted [%v], creating cluster and request service account",
 		request.Namespace, request.Name, cluster.Namespace, cluster.Name, status.Granted)
-	if status.Granted {
-		logrus.Debugf("Cluster registration request '%s/%s', creating registration secret", request.Namespace, request.Name)
-	}
 
-	// delete old clusterregistrations
-	crlist, _ := h.clusterRegistration.List(request.Namespace, metav1.ListOptions{})
-
-	for _, creg := range crlist.Items {
-		if creg.Spec.ClientID == request.Spec.ClientID && creg.Spec.ClientRandom != request.Spec.ClientRandom {
-			logrus.Infof("Deleting old clusterregistration %s/%s", creg.Namespace, creg.Name)
-			if err := h.clusterRegistration.Delete(creg.Namespace, creg.Name, nil); err != nil && !apierrors.IsNotFound(err) {
-				return nil, status, err
+	if status.Granted {
```

Comment:
What should happen if `!status.Granted`? Should the cluster registration, role bindings, etc, be created anyway?
In other words, should we return here if `!status.Granted`?

Reply:
Should be fine, but this is an optimization that doesn't necessarily belong into this PR? Or does it...

Reply:
Right, as discussed, let's handle that via a separate PR, no worries.